### PR TITLE
Add Auth backend

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
The `lpc55-hal` crate is missing support for the `PUF` so we cannot use it (for now).
The solution will be to update it. Starting point:

This struct doesn't have a PUF: https://docs.rs/lpc55-hal/latest/lpc55_hal/struct.Peripherals.html

This one has: https://docs.rs/lpc55-pac/latest/lpc55_pac/struct.Peripherals.html

So the PUF peripheral is lost [here](https://github.com/Nitrokey/nitrokey-3-firmware/blob/main/runners/embedded/src/soc_lpc55/mod.rs#L32) in the conversion.


The PUF is a bit more complex than the ER in the nrf. It first needs to be `enrolled`, which generates a 1192 "activation code that needs to be stored in the flash". Then a key needs to be generated, and the key code (52 bytes for 256 bits I think) needs to be stored too. 

Then using the same activation code and key code the key can be extracted.

It would actually be more secure to not extract and use the PUF key to encrypt the PIN to make extraction of data for offline attacks impossible.

**Comment by sosthene-nitrokey** [comment]
Relevant PR: https://github.com/lpc55/lpc55-hal/pull/61

File: Cargo.toml [hunk 1]
```diff
@@ -14,8 +14,8 @@ version = "1.2.2-alpha.20230224"
 # forked
 ctap-types = { git = "https://github.com/Nitrokey/ctap-types", tag = "v0.1.2-nitrokey.1" }
 littlefs2 = { git = "https://github.com/Nitrokey/littlefs2", tag = "v0.3.2-nitrokey-2" }
-lpc55-hal = { git = "https://github.com/Nitrokey/lpc55-hal", tag = "v0.3.0-nitrokey-1" }
-trussed = { git = "https://github.com/Nitrokey/trussed", tag = "v0.1.0-nitrokey.6" }
+lpc55-hal = { git = "https://github.com/robin-nitrokey/lpc55-hal", rev = "f49c9abaa3ef6805d265f7cda3fe1920c3a53aad" }
```
Comment (robin-nitrokey) [inline Cargo.toml:17]: Please create a tag in our fork for `lpc55-hal`.

**Review by robin-nitrokey — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (sosthene-nitrokey) [reply Cargo.toml:17]: I don't have write access to it

Comment (robin-nitrokey) [reply Cargo.toml]: AFAIS the tag does not include the `status` and `allow` methods.

**Comment by robin-nitrokey** [comment]
I think this one should be ready to merge once we drop the lpc55-specific changes.

**Comment by sosthene-nitrokey** [comment]
Which changes should we drop? To me the current changes are what we want no?

**Comment by robin-nitrokey** [comment]
Didn’t we decide to not use a hardware key on the LPC55 and instead reserve the PUF for future use?

**Review by robin-nitrokey — APPROVED** [review]

**Pull request merged**